hometask2/hometask2_1.js

```diff
@@ -0,0 +1,145 @@
+const express = require("express");
+const app = express();
+
+const bodyParser = require("body-parser");
+app.use(bodyParser.json() );
+app.use(bodyParser.urlencoded({
+  extended: true
+})); 
+
+
+const Joi = require("@hapi/joi");
+const validator = require("express-joi-validation").createValidator({});
+
+const schema = Joi.object({
+	login: Joi.string()
+		.alphanum()
+		.min(5)
+		.max(30)
+		.required(),
+
+	password: Joi.string()
+		.pattern(new RegExp("^[a-zA-Z0-9]{8,30}$"))
+		.required(),
+
+	age: Joi.number()
+		.integer()
+		.min(4)
+		.max(130)
+		.required(),
+});
+
+
+
+const usersMap = new Map();
```

Comment:
By creating variables outside of your routes and using them across the routes logic, you're violating the REST principles of having a stateless endpoint handlers.
Consider scoping everything you need to the individual routes and their handlers to comply with the REST protocol.